### PR TITLE
In-2472 Add lost_reason to deals.list

### PR DESCRIPTION
**Pull request by luisaterrabrito**

### CODE REVIEW

**Review by sanderbrugge — DISMISSED** [review]

**Review by EOSullivanBerlin — DISMISSED** [review]

**Review by steftrenson — DISMISSED** [review]

File: apiary.apib [hunk 1]
```diff
@@ -374,6 +374,8 @@ We list all backwards-compatible additions here. These are currently available i
 
 #### Latest
 
+- We added lost_reason to the response of to `deals.list`.
```
Comment (tiagobastos) [inline apiary.apib]: Other fields in the `Latest` section have backtick around them. 
For consistency `lost_reason` should have too.

File: src/changes-backwards-compatible.apib [hunk 2]
```diff
@@ -6,6 +6,8 @@ We list all backwards-compatible additions here. These are currently available i
 
 #### Latest
 
+- We added lost_reason to the response of to `deals.list`.
```
Comment (tiagobastos) [inline src/changes-backwards-compatible.apib]: Here too.

**Review by sanderbrugge — DISMISSED** [review]

**Review by tiagobastos — DISMISSED** [review]

**Review by EOSullivanBerlin — DISMISSED** [review]

File: src/03-deals/deals.apib [hunk 3]
```diff
@@ -91,6 +91,11 @@ Get a list of deals.
                 + created_at: `2017-05-09T11:25:11+00:00` (string)
                 + updated_at: `2017-05-09T11:30:58+00:00` (string)
                 + `web_url`: `https://app.teamleader.eu/sale_detail.php?id=6e7fe84d-d4b3-4723-abae-9bc082d8f65a` (string)
+                + lost_reason (object)
+                    + reason (object)
+                        + type: lost_reason (string)
```
Comment (dietrichm) [inline src/03-deals/deals.apib]: B: types are camel cased, so this should be `lostReason`.

IIRC this is also a requirement to have the lost reasons sideloaded.

File: src/03-deals/deals.apib [hunk 4]
```diff
@@ -91,6 +91,11 @@ Get a list of deals.
                 + created_at: `2017-05-09T11:25:11+00:00` (string)
                 + updated_at: `2017-05-09T11:30:58+00:00` (string)
                 + `web_url`: `https://app.teamleader.eu/sale_detail.php?id=6e7fe84d-d4b3-4723-abae-9bc082d8f65a` (string)
+                + lost_reason (object)
```
Comment (dietrichm) [inline src/03-deals/deals.apib]: PP: place the `lost_reason` _before_ meta data properties (`created_at`, `updated_at` and `web_url`).

File: src/changes-backwards-compatible.apib [hunk 5]
```diff
@@ -6,6 +6,8 @@ We list all backwards-compatible additions here. These are currently available i
 
 #### Latest
 
+- We added `lost_reason` to the response of to `deals.list`.
```
Comment (dietrichm) [inline src/changes-backwards-compatible.apib]: PP:
```suggestion
- We added `lost_reason` to the response of `deals.list`. This is composed of a `lostReason` object along with a more elaborate `remark`.
```

**Review by dietrichm — CHANGES_REQUESTED** [review]

**Review by sanderbrugge — DISMISSED** [review]

**Review by tiagobastos — DISMISSED** [review]

**Review by dietrichm — DISMISSED** [review]

**Review by EOSullivanBerlin — DISMISSED** [review]

**Review by lowiebenoot — DISMISSED** [review]

**Review by tiagobastos — DISMISSED** [review]

File: src/03-deals/deals.apib [hunk 6]
```diff
@@ -88,11 +88,11 @@ Get a list of deals.
...
                     + type: `dealSource` (string)
                     + id: `52a4bbb2-d196-48d7-8e43-9ec2f44d06f9` (string)
-                + lost_reason (object)
-                    + reason (object)
+                + lost_reason (object, nullable)
+                    + reason (object, nullable)
                         + type: lostReason (string)
                         + id: `4e81f3c4-7dca-44cb-b126-6158bec392a2` (string)
-                    + remark: `Too expensive` (string)
+                    + remark: `Too expensive` (string, nullable)
```
Comment (dietrichm) [inline src/03-deals/deals.apib]: Q: can it happen that `lost_reason` is provided, but both or either reason or remark are `null`?

Comment (dietrichm) [reply src/03-deals/deals.apib]: Explained by @luisaterrabrito IRL; if the reason is unknown, there is no `lostReason` set.

**Review by dietrichm — DISMISSED** [review]

**Review by sanderbrugge — DISMISSED** [review]

**Review by EOSullivanBerlin — DISMISSED** [review]

**Review by EOSullivanBerlin — APPROVED** [review]

**Review by sanderbrugge — APPROVED** [review]

**Review by lowiebenoot — APPROVED** [review]

**Review by tiagobastos — APPROVED** [review]

**Pull request merged**